### DISCUSSION TITLE
The trailing diamond doesn't use the inherited background color of its segment

### DISCUSSION THREAD

**Discussion by TheOnlyTails** [JanDeDobbeleer/oh-my-posh#1006] [Troubleshoot] [answered]
I'm trying to make a "rounder" segment at the end of my shell line, but for some reason it looks like this: 

![image](https://user-images.githubusercontent.com/65342367/135120389-df1fdf61-2bb6-47a0-8055-762eab1901c5.png)

The ending is supposed to be blue, not black, and I have no idea why it uses the default color instead of inheriting the background color of the previous segment.

Here's the code of the component:
```json
{
  "type": "text",
  "style": "diamond",
  "background": "inherit",
  "foreground": "inherit",
  "trailing_diamond": "\uE0B4",
  "properties": {
    "text": "\uE0BD"
  }
}
```

Here's the full theme: https://gist.github.com/0779bfac3d2c94ee69181b3a8fdd6459

**Comment by JanDeDobbeleer** [comment, marked as answer]
That's probably going to be a bug in the diamond logic. Or rather, not taking inheritance into account. I'll check.

**Reply by JanDeDobbeleer** [reply]
@TheOnlyTails this is an interesting one. As we set the background color as foreground and it results in `inherit`, it takes the foreground color of the parent rather than the background. That's a direct result of adding the logic to the overrides. Also only happens on diamond segments with inheritance. I don't want to leak everything everywhere so I'll give it some thought.

**Reply by TheOnlyTails** [reply]
Hmm, interesting, I didn't think of that. It also  appears to work fine in the previous version.

**Reply by JanDeDobbeleer** [reply]
@TheOnlyTails yes, that's because there we set the color earlier so that context was available.

**Reply by TheOnlyTails** [reply]
Ah, I see. Let me know if you need any help then.

**Reply by JanDeDobbeleer** [reply]
Fixed it and found another small issue I could resolve as well.

**Reply by TheOnlyTails** [reply]
That's wonderful! Thank you very much!

**Reply by JanDeDobbeleer** [reply]
https://github.com/JanDeDobbeleer/oh-my-posh/pull/1007

**Reply by TheOnlyTails** [reply]
Working great!
